File: rtadp-proto/Supervisor2.py
# Copyright (C) 2024 INAF
# This software is distributed under the terms of the BSD-3-Clause license
#
# Authors:
#
#    Andrea Bulgarelli <andrea.bulgarelli@inaf.it>
#
from Supervisor import Supervisor
from WorkerManager2 import WorkerManager2

class Supervisor2(Supervisor):
	def __init__(self, config_file="config.json", name="RTADP2"):
		super().__init__(config_file, name)

	def start_managers(self):
		indexmanager=0
		manager1 = WorkerManager2(indexmanager, self, "Rate")
		manager1.start()
		self.manager_workers.append(manager1)


	#to be reimplemented ####
	#Decode the data before load it into the queue. For "dataflowtype": "binary"
	def decode_data(self, data):
		return data

	#to be reimplemented ####
	#Open the file before load it into the queue. For "dataflowtype": "file"
	#Return an array of data and the size of the array
	def open_file(self, filename):
		f = [filename]
		return f, 1

File: c++/src/rtadp-proto/Supervisor2.cpp
#include "Supervisor2.h"

// Constructor
Supervisor2::Supervisor2(const std::string& config_file, const std::string& name)
    : Supervisor(config_file, name) {
}

// Destructor
Supervisor2::~Supervisor2() {
    for (WorkerManager* m : manager_workers)
        delete m;
}

// Override the start_managers method
void Supervisor2::start_managers() {
    int indexmanager = 0;
    WorkerManager* manager1 = new WorkerManager2(indexmanager, this, workername);
    // setup_result_channel(manager1, indexmanager);
    manager1->run();
    manager_workers.push_back(manager1);
    // TODO: Rimuovere print o meglio trasformare come log
    std::cout << "DER SUP manager started. manager_workers lenght: " << manager_workers.size() << std::endl;
}

// Decode the data before loading it into the queue. For "dataflowtype": "binary"
zmq::message_t& Supervisor2::decode_data(zmq::message_t& data) {
    return data;
}

// Open the file before loading it into the queue. For "dataflowtype": "file"
// Return an array of data and the size of the array
std::pair<std::vector<std::string>, int> Supervisor2::open_file(const std::string& filename) {
    std::vector<std::string> f = { filename };
    return { f, 1 };
}
